firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/InternalKeys.java

```diff
@@ -0,0 +1,98 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.crashlytics.internal.common;
+
+import androidx.annotation.NonNull;
+import com.google.firebase.crashlytics.internal.Logger;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/** Handles custom attributes internal to the SDK, eg. for custom Unity metadata. */
+public class InternalKeys {
```

Comment:
The implementations of `InternalKeys` and `UserMetadata` are almost identical. Did you consider making a `KeyValueStore` class and making `UserMetadata` extend from it? Then the `internalKeys` object is an instance of `KeyValueStore`. Usually I'm not too worried about DRY but I *think* in this case it makes more sense to use inheritance, since the two stores are so conceptually similar. Thoughts?